src/twisted/test/test_twistd.py

```diff
@@ -359,6 +360,27 @@ def test_badAttributeWithConfiguredLogObserver(self):
         self.assertNotIn('\n', e.args[0])
 
 
+    def test_version(self):
+        """
+        Print out the version.
+        """
+        from twisted import copyright
+
+        if platformType == "win32":
+            name = "(the Twisted Windows runner)"
+        else:
+            name = "(the Twisted daemon)"
+        expectedOutput = ('twistd {} {}\n{}\n'.format(
+            name, copyright.version, copyright.copyright))
+
+        out = NativeStringIO()
+        stdout = self.patch(sys, 'stdout', out)
+        config = twistd.ServerOptions()
```

Comment:
Are you talking about modifying ServerOptions?  There are two implementations,
one for Windows and one for Unix, so any modification would need to be made to both places.

Reply:
Yeah.  The change to the initializer could be done in [a shared base class](https://github.com/twisted/twisted/blob/487f9786d6cc01d5fc1cac5b815db19aed998d28/src/twisted/application/app.py#L592-L594), but all the `print` functions in each subclass would have to gain a new `file` argument.  Changing the initializer to accept a default argument should be safe even though this is a public API, because subclasses and callers don't need to change.

The problem with patching `sys.stdout` is that anything in the Python process can write to it, which means seemingly unrelated changes in far away parts of Twisted might render this test's assertions about the contents of `out` incorrect. So I think the change is worth it.

I would also prefer `checkPID's` signature to include `os.kill` as a default argument (e.g., `def checkPID(pidfile, kill=os.kill)`), but its code path is shallow enough that dependency injection seems less necessary.

Reply:
I added something to set stdout and pass it down to the print().